src/lib/reasoners/adt_rel.ml

```diff
@@ -558,75 +534,47 @@ let remove_redundancies la =
          cache := SLR.add a !cache;
          true
        end
-    )la
+    ) la
 
-let assume env uf la =
-  if Options.get_disable_adts () then
-    env, { Sig_rel.assume = []; remove = [] }
-  else
-    let la = remove_redundancies la in (* should be done globally in CCX *)
-    let delayed, result = Rel_utils.Delayed.assume env.delayed uf la in
-    let env = count_splits env la in
-    let classes = Uf.cl_extract uf in
-    let env = { env with classes = classes } in
-    let aux bol r1 r2 dep env eqs = function
-      | None     -> env, eqs
-      | Some hss ->
-        if bol then
-          add_eq uf hss (Th.embed r1) (Th.embed r2) dep env eqs
-        else
-          add_diseq uf hss (Th.embed r1) (Th.embed r2) dep env eqs
-    in
-    Debug.print_env "before assume" env;
-    let env, eqs =
-      List.fold_left
-        (fun (env,eqs) (a, b, c, d) ->
-           Debug.assume a;
-           match a, b, c, d with
-           | Xliteral.Eq (r1, r2), _, ex, _ ->
-             (* needed for models generation because fresh terms are not
-                added with function add *)
-             let env = add_rec (add_rec env r1) r2 in
-             aux true  r1 r2 ex env eqs (values_of (X.type_info r1))
-
-           | Xliteral.Distinct(false, [r1;r2]), _, ex, _ ->
-             (* needed for models generation because fresh terms are not
-                added with function add *)
-             let env = add_rec (add_rec env r1) r2 in
-             aux false r1 r2 ex env eqs (values_of (X.type_info r1))
-
-           | Xliteral.Builtin(true, Sy.IsConstr hs, [e]), _, ex, _ ->
-             assume_is_constr uf hs e ex env eqs
-
-           | Xliteral.Builtin(false, Sy.IsConstr hs, [e]), _, ex, _
-             [@ocaml.ppwarning "XXX: assume not (. ? .): reasoning missing ?"]
-             ->
-             assume_not_is_constr uf hs e ex env eqs
-
-           | _ -> env, eqs
-
-        ) (env,[]) la
-    in
-    let eqs = List.rev_append env.pending_deds eqs in
-    let env = {env with pending_deds = []} in
-    Debug.print_env "after assume" env;
-    let print fmt (a,_,_) =
-      match a with
-      | Literal.LTerm a -> Format.fprintf fmt "%a" E.print a;
-      | _ -> assert false
-    in
-    if Options.get_debug_adt () then
-      Printer.print_dbg ~module_name:"Adt_rel" ~function_name:"assume"
-        "assume deduced %d equalities@ %a" (List.length eqs)
-        (Printer.pp_list_no_space print) eqs;
-    let eqs = List.rev_append eqs result.assume in
-    { env with delayed }, { Sig_rel.assume = eqs; remove = [] }
+(* Update the counter of case-split size in [env]. *)
+let count_splits env la =
+  List.fold_left
+    (fun nb (_, _, _, i) ->
+       match i with
+       | Th_util.CS (Th_util.Th_adt, n) -> Numbers.Q.mult nb n
+       | _ -> nb
+    ) env.size_splits la
 
+let assume env uf la =
+  Debug.pp_env "before assume" env;
+  let la = remove_redundancies la in
```

Comment:
Please keep the comment about doing this globally in CCX which is still relevant.